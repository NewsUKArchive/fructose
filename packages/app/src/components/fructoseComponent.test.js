/* globals describe it beforeEach expect */
import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";
import { EventEmitter } from "events";
import FructoseComponent from "./fructoseComponent";

const MockComponent = React.createClass({
  render: () => <Text id={this.props.id} />
});

describe("Functional React Component Tester UI", () => {
  let wrapper;
  let events;

  beforeEach(() => {
    events = new EventEmitter();
    wrapper = shallow(
      <FructoseComponent
        events={events}
        components={{ "abc123-test-string": <MockComponent id="testIdBlah" /> }}
      />
    );
  });

  it("snapshot: instantiated FunctionalRCTUI component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("adds event listener for 'load' on creation", () => {
    expect(events.listeners("load")).toHaveLength(1);
  });

  it("loads default component on render", () => {
    expect(wrapper.contains(<Text>Fructose</Text>)).toBe(true);
  });

  it("snapshot: loading MockComponent", () => {
    wrapper.instance().loadComponent("abc123-test-string");
    expect(wrapper.update()).toMatchSnapshot();
  });

  it("loadComponent loads component into View", () => {
    wrapper.instance().loadComponent("abc123-test-string");
    expect(wrapper.update().contains(<MockComponent id="testIdBlah" />)).toBe(
      true
    );
  });

  it("loadComponent throws error when component not found", () => {
    expect(() =>
      wrapper.instance().loadComponent("non-existent-component")
    ).toThrow();
  });

  it("loadComponent throws error when component not found", () => {
    expect(() =>
      wrapper.instance().loadComponent("non-existent-component")
    ).toThrow();
  });

  it("removes 'load' listener on unmount", () => {
    wrapper.unmount();
    expect(events.listeners("load")).toHaveLength(0);
  });
});
