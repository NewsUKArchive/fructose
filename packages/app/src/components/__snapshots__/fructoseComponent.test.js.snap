// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional React Component Tester UI snapshot: instantiated FunctionalRCTUI component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Fructose
  </Text>
</View>
`;

exports[`Functional React Component Tester UI snapshot: loading MockComponent 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <MockComponent
    id="testIdBlah"
  />
</View>
`;
